typings/n-express.d.ts

```diff
@@ -0,0 +1,80 @@
+import Metrics from '@financial-times/n-metrics';
+import Express from 'express';
+import http from 'http';
+import https from 'https';
+import { cacheHeaders } from '../src/middleware/cache';
+export * from './metrics';
+
+export type CacheHeaders = typeof cacheHeaders;
+
+export interface Request extends Express.Request {}
+
+export interface Response extends Express.Response, CacheHeaders {
+	unvary: (...args: string[]) => void;
+	unvaryAll: () => void;
+
+	// Backwards-compatible uber-camel-cased names
+	unVary: unvary;
+	unVaryAll: unvaryAll;
+}
+
+export interface NextFunction extends Express.NextFunction {}
+
+export type Callback = (
+	req: Request,
+	res: Response,
+	next: NextFunction
+) => void;
+
+export interface AppMeta {
+	name: string;
+	directory: string;
+	systemCode: string;
+	graphiteName: string;
+}
+export interface AppOptions extends AppMeta {
+	healthChecksAppName?: string;
+	healthChecks: Metrics.Healthcheck[];
+	errorRateHealthcheck?: AppError;
+	demo?: boolean;
+	logVary?: boolean;
+	withAb: boolean;
+	withAnonMiddleware?: boolean;
+	withConsent: boolean;
+	withBackendAuthentication: boolean;
+	withFlags: boolean;
+	withServiceMetrics: boolean;
+}
+
+export interface AppError {
+	severity?: number;
+	threshold?: number;
+	samplePeriod?: string;
+}
```

Comment:
**suggestion**: the name `AppError` for this is potentially misleading, might imply it's an actual Error object. `ErrorRateHealthcheckOptions` would be clearer